lib/assert.js

```diff
@@ -122,7 +122,11 @@ function wrapAssertions(callbacks) {
 			} else if (isObservable(fn)) {
 				promise = observableToPromise(fn);
 			} else if (typeof fn !== 'function') {
-				throw new TypeError('t.throws must be called with a function, Promise, or Observable');
+				fail(this, new AssertionError({
+					actual: fn,
+					message: '`t.throws()` must be called with a function, Promise, or Observable'
+				}));
```

Comment:
`--fail-fast` implies all tests stop. This still immediately fails the test. The behavior should be similar to before this refactor, except that the error can no longer be caught by calling code.